test/csi/toolkit/counter_tool.cpp

```diff
@@ -1,28 +1,27 @@
+#include "csi.h"
 #include <stdio.h>
 #include <stdlib.h>
 
+extern "C" {
 long times_accessed_by_size[4];
 
-static void __csi_destroy() {
+WEAK void __csi_destroy() {
```

Comment:
That's true. In that case, it should probably not be named `__csi_destroy`, and shouldn't be in `csi.h` or the null tool. Instead, a tool should define its own destroy functions and register them via `atexit`. I'll make those changes.